core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RangeValueFilterExecuterImpl.java

```diff
@@ -0,0 +1,557 @@
+
+    if (isDimensionPresentInCurrentBlock == true) {
+      if (((lessThan == true) && (filterMaxLessThanBlockMin >= 0)) ||
+          ((lessThanEqualTo == true) && (filterMaxLessThanBlockMin > 0)) ||
+          ((greaterThan == true) && (filterMinGreaterThanBlockMax >= 0)) ||
+          ((greaterThanEqualTo == true) && (filterMinGreaterThanBlockMax > 0))) {
+        isScanRequired = false;
+      }
+      return isScanRequired;
+    } else {
+      return isDefaultValuePresentInFilter;
+    }
+  }
+
+  /**
+   * Method checks is the scan lies within the range values or not.
+   * @param blockMaxValue
+   * @param blockMinValue
+   * @return
+   */
+  @Override public BitSet isScanRequired(byte[][] blockMaxValue, byte[][] blockMinValue) {
+    BitSet bitSet = new BitSet(1);
+    byte[][] filterValues = this.filterRangesValues;
+    int columnIndex = this.dimColEvaluatorInfo.getColumnIndex();
+    boolean isScanRequired =
+        isScanRequired(blockMinValue[columnIndex], blockMaxValue[columnIndex], filterValues);
+    if (isScanRequired) {
+      bitSet.set(0);
+    }
+    return bitSet;
+  }
+
+  /**
+   * Method to apply the Range Filter.
+   * @param blockChunkHolder
+   * @return
+   * @throws FilterUnsupportedException
+   * @throws IOException
+   */
+  public BitSetGroup applyNoAndDirectFilter(BlocksChunkHolder blockChunkHolder)
+      throws FilterUnsupportedException, IOException {
+
+    // select all rows if dimension does not exists in the current block
+    if (!isDimensionPresentInCurrentBlock) {
+      int numberOfRows = blockChunkHolder.getDataBlock().nodeSize();
+      return FilterUtil
+          .createBitSetGroupWithDefaultValue(blockChunkHolder.getDataBlock().numberOfPages(),
+              numberOfRows, true);
+    }
+
+    int blockIndex = segmentProperties.getDimensionOrdinalToBlockMapping()
+        .get(dimColEvaluatorInfo.getColumnIndex());
+
+    if (null == blockChunkHolder.getDimensionRawDataChunk()[blockIndex]) {
+      blockChunkHolder.getDimensionRawDataChunk()[blockIndex] = blockChunkHolder.getDataBlock()
+          .getDimensionChunk(blockChunkHolder.getFileReader(), blockIndex);
+    }
+
+    DimensionRawColumnChunk rawColumnChunk =
+        blockChunkHolder.getDimensionRawDataChunk()[blockIndex];
+    BitSetGroup bitSetGroup = new BitSetGroup(rawColumnChunk.getPagesCount());
+    for (int i = 0; i < rawColumnChunk.getPagesCount(); i++) {
+      if (rawColumnChunk.getMaxValues() != null) {
+        if (isScanRequired(rawColumnChunk.getMinValues()[i], rawColumnChunk.getMaxValues()[i],
+            this.filterRangesValues)) {
+          if (isRangeFullyCovered(rawColumnChunk.getMinValues()[i],
+              rawColumnChunk.getMaxValues()[i], this.filterRangesValues)) {
+            // Set all the bits in this case as filter Min Max values cover the whole block.
+            BitSet bitSet = new BitSet(rawColumnChunk.getRowCount()[i]);
+            bitSet.flip(0, rawColumnChunk.getRowCount()[i]);
```

Comment:
Here bitSet.flip parameters are startIndex and endIndex of the bits to flip. The new BitSet will give all bit initialized to 0. bitSet.flip will turn into 1.